File: DesafioMobile/app/src/main/java/br/com/tiago/desafiomobile/ui/fragment/TransactionFragment.java
package br.com.tiago.desafiomobile.ui.fragment;

import android.os.Bundle;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import java.util.List;

import br.com.tiago.desafiomobile.R;
import br.com.tiago.desafiomobile.interactor.impl.TransactionInteractorImpl;
import br.com.tiago.desafiomobile.model.TransactionModel;
import br.com.tiago.desafiomobile.presenter.TransactionPresenter;
import br.com.tiago.desafiomobile.presenter.impl.TransactionPresenterImpl;
import br.com.tiago.desafiomobile.ui.adapter.TransactionRecyclerAdapter;
import butterknife.BindView;
import butterknife.ButterKnife;

/**
 * Created by TiagoCabral on 8/22/2017.
 */

public class TransactionFragment extends BaseFragment {

    private static String TAG = TransactionFragment.class.getSimpleName();
    private TransactionRecyclerAdapter transactionRecyclerAdapter;
    private static TransactionPresenter transactionPresenter;

    @BindView(R.id.transaction_recycler_view)
    RecyclerView recyclerView;

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        Log.d(TAG, "onCreateView: inicio");
        View inflate;

        inflate = inflater.inflate(R.layout.transaction_history_fragment, container, false);
        ButterKnife.bind(this, inflate);

        transactionRecyclerAdapter = new TransactionRecyclerAdapter();
        recyclerView.setAdapter(transactionRecyclerAdapter);
        recyclerView.setLayoutManager(new LinearLayoutManager(getActivity()));
        transactionPresenter = new TransactionPresenterImpl(this, new TransactionInteractorImpl(getContext()));

        return inflate;
    }


    public static void getTransactions() {
        transactionPresenter.getTransactions();
    }

    public void addAll(List<TransactionModel> transactionModels) {
        transactionRecyclerAdapter.addAll(transactionModels);
    }
}

File: app/src/main/java/com/example/pharol/temminstore/transaction/TransactionFragment.kt
package com.example.pharol.temminstore.transaction

import android.arch.lifecycle.LifecycleFragment
import android.arch.lifecycle.Observer
import android.arch.lifecycle.ViewModelProviders
import android.os.Bundle
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.example.pharol.temminstore.R
import com.example.pharol.temminstore.TemminApplication
import com.example.pharol.temminstore.di.ui.DaggerFragmentComponent
import com.example.pharol.temminstore.di.ui.FragmentModule

class TransactionFragment: LifecycleFragment() {

    val component by lazy {
        DaggerFragmentComponent.builder().fragmentModule(FragmentModule(this)).
                applicationComponent((this.context.applicationContext as TemminApplication).applicationComponent).
                build()
    }

    @javax.inject.Inject lateinit var transactionViewFactory : TransactionViewModelFactory

    lateinit var transactionView: TransactionViewModel

    val transactionAdapter = TransactionRecyclerAdapter(arrayListOf())
    var recyclerView: RecyclerView? = null

    override fun onCreate(savedInstanceState: android.os.Bundle?) {
        super.onCreate(savedInstanceState)
        component.inject(this)
        transactionView = ViewModelProviders.of(this, transactionViewFactory).get(TransactionViewModel::class.java)
    }

    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val returnView = inflater?.inflate(R.layout.fragment_transactions, container, false)

        recyclerView = returnView?.findViewById(R.id.rv_list_transactions) as RecyclerView
        recyclerView?.layoutManager = android.support.v7.widget.LinearLayoutManager(activity)
        recyclerView?.adapter = transactionAdapter

        return returnView
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)

        transactionView.getTransactions().observe(this, Observer {
            if (it != null) {
                transactionAdapter.listTransactions = it
            }
            recyclerView?.adapter?.notifyDataSetChanged()
        })

    }
}